app/helpers/automated_tests_helper.rb

```diff
@@ -380,6 +474,435 @@ def parse_test_output(repo_assignment_dir, assignment, logfile, data)
+    all_scripts = TestScript.where(assignment_id: assignment.id)
+
+    list_run_scripts = Array.new
+
+    # If the test run is requested at collection (by Admin or TA),
+    # All of the test scripts should be run.
+    if call_on == "collection"
+      list_run_scripts = all_scripts
+    else
+      # If the test run is requested at submission or upon request,
+      # verify the script is allowed to run.
+      all_scripts.each do |script|
+        if (call_on == "submission") && script.run_on_submission
+          list_run_scripts.insert(list_run_scripts.length, script)
+        elsif (call_on == "request") && script.run_on_request
+          list_run_scripts.insert(list_run_scripts.length, script)
+        end
+      end
+    end
+
+    # sort list_run_scripts using ruby's in place sorting method
+    list_run_scripts.sort_by! {|script| script.seq_num}
+
+    # list_run_scripts should be sorted now. Perform a check here.
+    # Take this out if it causes performance issue.
+    ctr = 0
+    while ctr < list_run_scripts.length - 1
+      if (list_run_scripts[ctr].seq_num) > (list_run_scripts[ctr+1].seq_num)
+        raise "list_run_scripts is not sorted"
+      end
+      ctr = ctr + 1
+    end
+
+    return list_run_scripts
+  end
+
+  # Request an automated test. Ask Resque to enqueue a job.
+  def self.async_test_request(grouping_id, call_on)
+    if files_available?
+      if has_permission?
+        Resque.enqueue(AutomatedTestsHelper, grouping_id, call_on)
+      end
+    end
+  end
+
+
+  # Verify that MarkUs has some files to run the test.
+  # Note: this does not guarantee all required files are presented.
+  # Instead, it checks if there is at least one test script and
+  # source files are successfully exported.
+  def self.files_available?()
+    test_dir = File.join(MarkusConfigurator.markus_config_automated_tests_repository, @assignment.short_identifier)
+    src_dir = @repo_dir
+    assign_dir = @repo_dir + "/" + @assignment.repository_folder
+    if !(File.exists?(test_dir))
+      # TODO: show the error to user instead of raising a runtime error
+      raise I18n.t("automated_tests.test_files_unavailable")
+    elsif !(File.exists?(src_dir))
+      # TODO: show the error to user instead of raising a runtime error
+      raise I18n.t("automated_tests.source_files_unavailable")
+    end
+
+    if !(File.exists?(assign_dir))
+      # TODO: show the error to user instead of raising a runtime error
+      raise I18n.t("automated_tests.source_files_unavailable")
+    end
+
+    dir_contents = Dir.entries(assign_dir)
+
+    #if there are no files in repo (ie only the current and parent directory pointers)
+    if (dir_contents.length <= 2)
+      raise I18n.t("automated_tests.source_files_unavailable")
+    end
+
+    scripts = TestScript.where(assignment_id: @assignment.id)
+    if scripts.empty?
+      # TODO: show the error to user instead of raising a runtime error
+      raise I18n.t("automated_tests.test_files_unavailable")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.